### ISSUE TITLE
Yaw wrap-around causes velocity spikes

### ISSUE THREAD

**Issue by irwineffect** [PalouseRobosub/robosub-2018#329]
When the control system calculates its orientation velocity, we have an issue where if the yaw wraps around (e.g. -180 to 180), we get a spike in our velocity. Below is an image showing it occuring in simulation:
<img width="640" alt="yaw_velocity_error" src="https://user-images.githubusercontent.com/3589531/32253131-8d238b5c-be56-11e7-9a26-031f8734ba48.png">

This will cause problems if we use the derivative term for our yaw PID.

**Comment by daymare** [comment]
We can wrap around the previous value and take the smallest distance from both the previous value and the wrapped value.

```c
int wrap(int value)
{
    if (value > 0)
        return value - 360;
    else
        return value + 360;
}

int getDelta(previous, current)
{
    int wrap_prev = wrap(previous);
    
    if (abs(current-prev) < abs(current-wrap_prev))
       return current-prev
    else
        return current-wrap_prev
}
```

**Comment by ryan-summers** [comment]
So I did some testing and this actually isn't an error for the D-term. Remember that we aren't taking the derivative of the yaw to get yaw velocity in the control system, but actually the derivative of the yaw's _error_. The error value is properly wrapped, so we don't see these spikes. I've attached a demo of the yaw wrapping with the proper derivative shown (note that this is after I have implemented a fix for derivative calculations as mentioned in https://github.com/PalouseRobosub/robosub/issues/330). 
![yaw-wrap](https://user-images.githubusercontent.com/8771450/34535624-56ae9c44-f077-11e7-8485-23107f1d9328.png)
